### PR TITLE
[tests]: Add SecurityContext to support more environments

### PR DESCRIPTION
**Pull request by Barakmor1**
Set `Privileged`, `AllowPrivilegeEscalation`, `RunAsNonRoot`, `SeccompProfile`  and `Capabilities` in the SecurityContext to improve compatibility across environments and avoid warnings or errors related
to security policies.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: tests/e2e_application_aware_resource_quota_test.go [hunk 1]
```diff
@@ -1968,6 +1997,17 @@ func newTestPodWithAffinityForQuota(f *framework.Framework, name string, affinit
...
+					SecurityContext: &v1.SecurityContext{
+						Privileged:               ptr.To(false),
+						AllowPrivilegeEscalation: ptr.To(false),
+						RunAsNonRoot:             ptr.To(true),
+						SeccompProfile: &v1.SeccompProfile{
+							Type: v1.SeccompProfileTypeRuntimeDefault,
+						},
+						Capabilities: &v1.Capabilities{
+							Drop: []v1.Capability{"ALL"},
+						},
```
Comment (iholder101) [inline tests/e2e_application_aware_resource_quota_test.go:2009]: nit: WDYT about creating a helper function to reduce duplications? Something like `generateUnprivilegedSecurityContext()`?

**Review by iholder101 — COMMENTED** [review]
Looks good!
Left one nit, feel free to ignore
/lgtm

File: tests/e2e_application_aware_resource_quota_test.go (hunk 1 above)
Comment (Barakmor1) [reply tests/e2e_application_aware_resource_quota_test.go:2009]: Hey, thanks for the review! I'm actually planning a follow-up PR for refactoring soon, and I'll definitely focus on improving the pod creation then.

**Pull request merged**